core/sdk/src/main/java/com/mongodb/stitch/core/auth/internal/CoreStitchAuth.java

```diff
@@ -267,9 +297,15 @@ private synchronized Response doAuthenticatedRequest(
   @Override
   public <T> Stream<T> openAuthenticatedStream(final StitchAuthRequest stitchReq,
                                                final Decoder<T> decoder) {
-    if (!isLoggedIn()) {
-      throw new StitchClientException(StitchClientErrorCode.MUST_AUTHENTICATE_FIRST);
+    try {
+      if (!isLoggedInInterruptibly()) {
+        throw new StitchClientException(StitchClientErrorCode.MUST_AUTHENTICATE_FIRST);
+      }
+    } catch (InterruptedException e) {
+      System.err.println("Stream authentication was interrupted.");
+      return null;
```

Comment:
[q] are we sure we want to return this? why not just throw the exception? This seems like a null pointer exception waiting to happen somewhere upstream. What happens if the user calls `watch` and the stream got interrupted? I think it would be better for an error to happen rather than getting a null result.

Reply:
At the moment, that would have to happen on the users accord. We should probably throw an exception.